packages/flutter/lib/fix_data.yaml

```diff
@@ -14,6 +14,26 @@
 
 version: 1
 transforms:
+  # Changes made in https://github.com/flutter/flutter/pull/96109
+  - title: "Migrate `hashValues` to `Object.hash`"
+    date: 2022-01-07
+    element:
+      uris: [ 'material.dart', 'dart:ui' ]
+      function: 'hashValues'
+    changes:
+      - kind: 'rename'
+        newName: 'Object.hash'
+
+  # Changes made in https://github.com/flutter/flutter/pull/96109
+  - title: "Migrate `hashList` to `Object.hashAll`"
+    date: 2022-01-07
+    element:
+      uris: [ 'material.dart', 'dart:ui' ]
+      function: 'hashList'
+    changes:
+      - kind: 'rename'
+        newName: 'Object.hashAll'
+
```

Comment:
I'm a little worried that this is going to break people's code because of hashList taking nulls and hashAll not taking nulls. Is there a way to do this that is guaranteed safe?

cc @Piinks

Reply:
AFAIK, if the user was using `hashList` with ` null`, dart fix cannot fix that for them when it changes to `hashAll`. @bwilkerson would know for sure though.

Reply:
Any update on this?

Reply:
We have other dart fixes that are partial fixes, like if `SomeAPI.oldValue` was a `bool` and `SomeAPI.newValue` was an `enum`, so I think this is ok.
The fix tool will bring users to the new API, and they may need to update a little bit after the tool makes the migration.

Reply:
I am wondering if these fix rules will work though, since there is no class specified. Typically you would have to write a fix for every single class. These fix rules do not have tests yet, so we'll know for sure once tests are added.

They are in the test_fixes directory.

Reply:
Sorry for the delay; I missed the original mention.

No, I don't believe that we currently have any support for dealing with a change from a nullable parameter to a non-nullable parameter. In theory we could automate that, but it isn't clear to me what edit we'd want to apply. I can think of three semi-reasonable possibilities:
- Add a null check operator (`!`) after the argument if the argument is nullable. This would convert the error to a runtime error rather than a compile time error. I'm generally opposed to adding such operators automatically because it just hides the fact that the user needs to think about whether this is the right change.
- Refuse to change any invocation in which the argument is nullable. This would prevent us from making the rest of the changes, but might actually be better in some cases because some users might prefer to replace the invocation with something completely different in such cases.
- Do what we're doing today and require the user to manually fix the resulting error.

None of them are ideal, but I'm kind of leaning toward the last one because I suspect that it's doing as much for users as we reasonably can do.